cpp/proto/substrait/extension_rels.proto

```diff
@@ -44,3 +44,17 @@ message AsOfJoinRel {
     repeated .substrait.Expression by = 2;
   }
 }
+
+// Named tap relation
+//
+// A tap is a relation having a single input relation that it passes through, while also
+// causing some side-effect, e.g., writing to external storage.
+message NamedTapRel {
+  // The kind of tap
+  string kind = 1;
+  // A name used to configure the tap, e.g., a URI defining the destination of writing
+  string name = 2;
+  // Column names for the tap's output. If specified there must be one name per field.
+  // If empty, field names will be automatically generated.
```

Comment:
Why don't we make the columns required - i.e., if it's missing or doesn't match the number of fields in the data then it's an illegal message?

Reply:
I think we should use 
https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto#L268

To define the named tal rel similar to how we implement Asof join extension with MultiRelExtension

Reply:
I have no strong opinion on this either.  I do agree there will be cases where the tap doesn't care about the names but forcing the producer to invent field names in this case isn't much burden.